frontend/src/hooks/api/getters/useFeatureSearch/useFeatureSearch.ts

```diff
@@ -24,6 +24,19 @@ const fallbackData: SearchFeaturesSchema = {
     total: 0,
 };
 
+/**
+ With dynamic search and filter parameters we want to prevent cache from growing extensively.
+ We only keep the latest cache key `currentKey` and remove all other entries identified
+ by the `clearPrefix`
+ */
+const useClearSWRCache = (currentKey: string, clearPrefix: string) => {
```

Comment:
it can be extracted to a first class utility once we find it useful in other places